### ISSUE TITLE
Error while running train_seg.sh

### ISSUE THREAD

**Issue by won-bae** [GuoleiSun/MCIS_wsss#13]
Hi,

Thanks for sharing a great work. I have a question about training a segmentation algorithm.
While running train_seg.sh, I ran into 
Check failed: status == CUBLAS_STATUS_SUCCESS (11 vs. 0)  CUBLAS_STATUS_MAPPING_ERROR

Some solutions such as (https://stackoverflow.com/questions/39506837/segnet-cublas-status-success-11-vs-0-cublas-status-mapping-error) suggests to change the channel size of labels from 3 to 1.

If my understanding is correct, in MCIS, labels correspond to pseudo GTs created using get_gt.py and when I checked the pseudo gt images, it does not have a channel dimension which I think is desirable.

At this point, I have no clue what causes the error. Have you run into the same error before? Any suggestions would be appreciated!

**Comment by GuoleiSun** [comment]
Hi, thanks for your interest. I didn't see this error before. You can try the follows:
1. Did you run the test successfully? Make sure the caffe is successfully installed.
2. After you generate pseudo GTs using get_gt.py, you need to convert the mask into grey scale by using convert.py (https://github.com/GuoleiSun/MCIS_wsss/blob/master/Classifier/convert.py), because we are using caffe-based deeplab.

**Comment by won-bae** [comment]
Hi Guolei,

Thank you so much for your email. Apologize for the late reply. It took me
sometime to try your second suggestion.
Indeed, it works! But unfortunately, I have got only 64.78 (for
multi-scale) which is about 1.8 points lower than what you reported.
To reproduce the number you reported in the paper,

   1. Can I just use the solver_res.prototxt as it is? I am asking this
   because there are some comments in
   https://github.com/GuoleiSun/MCIS_wsss/blob/master/segmentation/examples/seg/exp2/solver_res.prototxt
   I wonder if I need to change something for example stepsize.
   2. I have stored the checkpoint as every 1000steps. Since it is hard to
   check every snapshot, can you give me a tip about which step gives the best
   performance?

Thank you again for your help.
Looking forward to hearing from you.

Best regards,
Won Bae

**Comment by GuoleiSun** [comment]
Hi Won, that's strange. Did you follow the default parameters? 
For your question 1, you don't need to change the stepsize.
For your question 2, usually the later models (after 11000 epochs) can give good performance. You can try to conduct model selection on those models on val set and then test it on test set.

**Comment by won-bae** [comment]
Hi GuoleiSun,

Thank you for your quick reply.
I indeed follow the default parameters and will definitely run some
model selection on the val set.
Besides it, would it be possible for you to share the pseudo labels? It
would help a lot to reproduce the result you had!

Best regards,
Won Bae

**Comment by won-bae** [comment]
I have trained the segmentation network many times with the pseudo labels I generated using the code you provided and pretrained model. Unfortunately, the gap is still more than 1 point in mIoU which I don't think is due to the randomness.
If it isn't too hard for you, it would be very helpful if you provide the pseudo labels you used, or the gen_gt.py with the right parameters (all the parameters such as conflict, fg_thr etc. do not match with the comments).
Look forward to hearing from you. Thank you.
